plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/delete/DeletionVectors.java

```diff
@@ -0,0 +1,134 @@
+
+    private DeletionVectors() {}
+
+    public static Roaring64NavigableMap readDeletionVectors(TrinoFileSystem fileSystem, Location location, DeletionVectorEntry deletionVector)
+            throws IOException
+    {
+        if (deletionVector.storageType().equals(UUID_MARKER)) {
+            TrinoInputFile inputFile = fileSystem.newInputFile(location.appendPath(toFileName(deletionVector.pathOrInlineDv())));
+            ByteBuffer buffer = readDeletionVector(inputFile, deletionVector.offset().orElseThrow(), deletionVector.sizeInBytes());
+            Roaring64NavigableMap bitmaps = deserializeDeletionVectors(buffer);
+            if (bitmaps.getLongCardinality() != deletionVector.cardinality()) {
+                throw new TrinoException(DELTA_LAKE_INVALID_SCHEMA, "The number of deleted rows expects %s but got %s".formatted(deletionVector.cardinality(), bitmaps.getLongCardinality()));
+            }
+            return bitmaps;
+        }
+        if (deletionVector.storageType().equals(INLINE_MARKER) || deletionVector.storageType().equals(PATH_MARKER)) {
+            throw new TrinoException(NOT_SUPPORTED, "Unsupported storage type for deletion vector: " + deletionVector.storageType());
+        }
+        throw new IllegalArgumentException("Unexpected storage type: " + deletionVector.storageType());
+    }
+
+    public static String toFileName(String pathOrInlineDv)
+    {
+        int randomPrefixLength = pathOrInlineDv.length() - Base85Codec.ENCODED_UUID_LENGTH;
+        String randomPrefix = pathOrInlineDv.substring(0, randomPrefixLength);
+        checkArgument(ALPHANUMERIC.matchesAllOf(randomPrefix), "Random prefix must be alphanumeric: %s", randomPrefix);
+        String prefix = randomPrefix.isEmpty() ? "" : randomPrefix + "/";
+        String encodedUuid = pathOrInlineDv.substring(randomPrefixLength);
+        UUID uuid = decodeUuid(encodedUuid);
+        return "%sdeletion_vector_%s.bin".formatted(prefix, uuid);
+    }
+
+    public static ByteBuffer readDeletionVector(TrinoInputFile inputFile, int offset, int expectedSize)
+            throws IOException
+    {
+        byte[] bytes = new byte[expectedSize];
+        try (DataInputStream inputStream = new DataInputStream(inputFile.newStream())) {
+            checkState(inputStream.skip(offset) == offset);
+            int actualSize = inputStream.readInt();
+            if (actualSize != expectedSize) {
+                throw new TrinoException(DELTA_LAKE_INVALID_SCHEMA, "The size of deletion vector %s expects %s but got %s".formatted(inputFile.location(), expectedSize, actualSize));
+            }
+            inputStream.readFully(bytes);
+        }
+        return ByteBuffer.wrap(bytes).order(LITTLE_ENDIAN);
+    }
+
+    private static Roaring64NavigableMap deserializeDeletionVectors(ByteBuffer buffer)
+            throws IOException
+    {
+        checkArgument(buffer.order() == LITTLE_ENDIAN, "Byte order must be little endian: %s", buffer.order());
```

Comment:
can `deserializeDeletionVectors` take `byte[]` and wrap with `ByteBuffer` (or maybe `Slice`) locally?